rust/src/dalek.rs

```diff
@@ -0,0 +1,58 @@
+//! [PrivateKey] and [PublicKey] implementations for types in [x25519_dalek].
+//!
+//! Usage example:
+//!
+//! ```
+//! use x25519_dalek::{PublicKey, StaticSecret};
+//! use glome::{tag,verify};
+//! let raw_private_key = [15u8; 32]; // <- our private key loaded from somewhere
+//! let private_key: StaticSecret = raw_private_key.into();
+//! let raw_public_key = [16u8; 32]; // <- their public key loaded from somewhere
+//! let public_key: PublicKey = raw_public_key.into();
+//! tag(&private_key, &public_key, 0u8, b"Hello, world!");
+//! verify(&private_key, &public_key, 42u8, b"Goodbye.", b"\x00\x01\x02\x03");
```

Comment:
Can we make the verify take the actual tag result from tag, i.e. make this into an example that may roughly pass? Maybe bonus points to generate a key rather than picking 15 and 16? That 15 made my eyes twitch.

Reply:
Changed to the test vectors and verifying the generated tag. Not sure if I want to randomly generate a key in the doctests...